pkg/scheduler/api/unschedule_info.go

```diff
@@ -82,12 +82,28 @@ func GetJobOverMaxAllowedMessageForQueue(
 		queueName, resourceNameStr, details)
 }
 
-func GetGangEvictionMessage(taskNamespace, taskName string, minimum int32) string {
+func GetGangEvictionMessage(task *pod_info.PodInfo, job *podgroup_info.PodGroupInfo) string {
+	if len(job.GetSubGroups()) == 1 {
+		if defaultSubgroup, found := job.GetSubGroups()[podgroup_info.DefaultSubGroup]; found {
+			return getGangEvictionForDefaultSubgroupMessage(task.Namespace, task.Name, defaultSubgroup.GetMinAvailable())
+		}
+	}
+	subGroup := job.GetSubGroups()[task.SubGroupName]
```

Comment:
Maybe worth checking if subGroup was found so it won't panic in case there is a problem?

Reply:
I think the user will not know what it means. I suggested a small enhancemenet.